cirq-core/cirq/sim/simulator.py

```diff
@@ -706,6 +705,33 @@ def _create_act_on_args(
         Returns:
             The `TSimulatorState` for this simulator.
         """
+        raise NotImplementedError()
+
+    def _create_simulation_state(
+        self, initial_state: Any, qubits: Sequence['cirq.Qid']
+    ) -> TSimulatorState:
+        """Creates the state for a simulator.
+
+        Custom simulators should implement this method.
+
+        Args:
+            initial_state: The initial state for the simulation. The form of
+                this state depends on the simulation implementation. See
+                documentation of the implementing class for details.
+            qubits: Determines the canonical ordering of the qubits. This
+                is often used in specifying the initial state, i.e. the
+                ordering of the computational basis states.
+
+        Returns:
+            The `TSimulatorState` for this simulator.
+        """
+        _compat._warn_or_error(
```

Comment:
Look like these function bodies got swapped - shouldn't this error-and-redirect be in `_create_act_on_args` instead?

Reply:
No, our simulation code now calls _create_simulatiin_state. So we need to ensure for 3rd party simulators that have implemented _create_act_on_args, that we fall back to those, with a warning.

Reply:
Making sure I understand:
- Cirq simulations call `_create_simulation_state`
- Cirq `SimulationState` types override `_create_simulation_state`, and thus don't hit this error
- External `ActOnArgs` types _don't_ override `_create_simulation_state`, hit this error, and redirect to `_create_act_on_args`, which they do define.

Is that right? If so, I think the error message should be clearer that we are falling back on `_create_act_on_args` (it was not called directly), and the external type must override `_create_simulation_state`.

Reply:
That's correct. Yes, `_create_*` is never meant to be used or called by third-parties. It's just supposed to be implemented in simulators (first or third party) so that our simulation infrastructure knows how to create the specific simulators' states.

I'll make the error message more clear, and also update the code comment to make sure it gets cleaned up correctly in v0.16.

Reply:
Done